moma_controllers/panda_mpc/src/panda_mpc.cpp

```diff
@@ -28,14 +28,20 @@ bool PandaMpcController::init(hardware_interface::RobotHW* robot_hw,
   if (!init_franka_interfaces(robot_hw)) return false;
 
   std::string arm_description;
-  if (!node_handle.getParam("/ocs2_mpc/robot_description_ocs2", arm_description)) {
-    ROS_ERROR("Failed to retrieve /ocs2_mpc/robot_description_ocs2 from param server.");
+  if (!node_handle.getParam("/arm_description", arm_description)) {
+    ROS_ERROR("Failed to retrieve /arm_description from param server.");
     return false;
   }
   robot_model_ = std::make_unique<rc::RobotWrapper>();
   robot_model_->initFromXml(arm_description);
-  position_current_ = Eigen::VectorXd::Zero(robot_model_->getDof());
-  velocity_current_ = Eigen::VectorXd::Zero(robot_model_->getDof());
+
+  // the pinocchio model contains only the arm and not the full base which
+  // would require to maintain the full chain  of links
+  position_current_model_.setZero(robot_model_->getDof());
+  velocity_current_model_.setZero(robot_model_->getDof());
+  
+  position_current_ = Eigen::VectorXd::Zero(ocs2::mobile_manipulator::STATE_DIM(armInputDim_));
+  velocity_current_ = Eigen::VectorXd::Zero(ocs2::mobile_manipulator::STATE_DIM(armInputDim_));
```

Comment:
This is wrong then. `STATE_DIM` is the size of the full model, not only the arm.
I also wonder if this needs more changes down the line towards ocs2, since many things there depend on `STATE_DIM` and `INPUT_DIM`, which is now oversized.

Reply:
Maybe there is a misunderstanding. Here we use pinocchio in 2 places: 
- inside ocs2 for the full kinematic chain (3 base dofs and no gripper)
- in the simulation controller to get gravity compensation

My changes affect the second use. The "model-for-gravity-compensation" only uses the arm urdf, the rest stay unchanged as it was before.